packages/react-inbox/src/components/Inbox/index.tsx

```diff
@@ -82,8 +84,46 @@ const Inbox: React.FunctionComponent<InboxProps> = (props) => {
   }, []);
 
   useEffect(() => {
-    inbox.init(props);
-  }, [props]);
+    if (clientKey && userId) {
+      const localStorageState = localStorage.getItem(
+        `${clientKey}/${userId}/inbox`
+      );
+
+      if (localStorageState) {
+        try {
+          initInbox({
+            ...JSON.parse(localStorageState),
+            config: props,
+          });
+          return;
+        } catch (ex) {
+          console.log("error", ex);
+          // do nothing
+        }
+      }
+    }
+
+    initInbox({
+      config: props,
+    });
+  }, [props, clientKey, userId]);
```

Comment:
Should this be getting called everytime the props object changes?

Reply:
yeah, because we need to update the context